demos/common/cpp/utils/include/utils/shared_blob_allocator.h

```diff
@@ -0,0 +1,32 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#pragma once
+#include "ie_allocator.hpp"
+#include "opencv2/core.hpp"
+
+class SharedBlobAllocator : public InferenceEngine::IAllocator {
```

Comment:
Could it be a template class?

Reply:
there will be differences is alloc/lock functions if we use random type for input data. So I don't see any way how template could help us here.